content/en/docs/components/pipelines/author-a-pipelines/bring_your_own_container.md

```diff
@@ -0,0 +1,184 @@
++++
+title = "Building components using your own container"
+description = "Bring user-controlled container into component authoring in KFP v2"
+weight = 50
++++
+
+Container-based components make it easier to use the pre-existed container to build a KFP component. 
+The user will control the container entrypoint and is able to define the component by specifying their container, commands, and args in a `ContainerSpec` object with similar authoring experiences as defining Python components. 
+
+
+## Before you begin
+
+1. Run the following command to install the Kubeflow Pipelines SDK v(TODO) or higher. If you run this command in a Jupyter notebook, restart the kernel after installing the SDK. 
+
+```python
```

Comment:
nit: shell or bash?

Reply:
It should be bash. Thanks for the catch!